### PR TITLE
updated-readme

### PR DESCRIPTION
**Pull request by pennyyang17**
Made the following changes:
- Added information regrading our API endpoints: CRUD and List
- Added instructions with Running the Tests and Running the Service from lab-flask-tdd
- Added a short introduction to our promotions microservice

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +64,124 @@ tests/                     - test cases package
...
+- **Description**: Creates a new promotion.
+-   **Response**: Returns the created promotion details with a unique promotion ID.
+
+### Read a Promotion
+
+-   **Method**: `GET`
+-   **Description**: Retrieves details of a specific promotion by its ID.
+- **Response**: Returns the detailed information about the specific promotion being requested.
+### Update a Promotion
+
```
Comment (yc6714) [inline README.md:107]: I'm not sure why the URL is missed in this commit. Could you add -   **URL**: `/promotions/<promotion_id>` back ?

File: README.md [hunk 2]
```diff
@@ -58,6 +64,124 @@ tests/                     - test cases package
...
+-   **Method**: `PUT`
+-   **Description**: Updates an existing promotion.
+-   **Response**: Returns the updated promotion details.
+
+### Delete a Promotion
+
```
Comment (yc6714) [inline README.md:113]: same here **URL**: `/promotions/<promotion_id>`

File: README.md [hunk 3]
```diff
@@ -58,6 +64,124 @@ tests/                     - test cases package
...
+-   **Method**: `DELETE`
+-   **Description**: Deletes a promotion by its ID.
+-   **Response**: Returns a success message.
+
+### List Promotions
+
```
Comment (yc6714) [inline README.md:119]: **URL**: `/promotions`

File: README.md [hunk 4]
```diff
@@ -58,6 +64,124 @@ tests/                     - test cases package
...
+    }` 
+
+### Create a Promotion
+
+- **Method**: `POST`
+- **Description**: Creates a new promotion.
+-   **Response**: Returns the created promotion details with a unique promotion ID.
+
+### Read a Promotion
+
```
Comment (yc6714) [inline README.md:102]: Could you also add the URLs for the 'Create' and 'Read' operations? I would appreciate it!

**Review by yc6714 — APPROVED** [review]

**Review by raghavm1 — APPROVED** [review]
LGTM! Thank you Penny!

**Pull request merged**